File: lib/models/remedio_model.php
class Remedio {
public $nome;
public $data;
public $id;
public $pet;

public function __construct($nome, $data, $id, $pet) {
$this->nome = $nome;
$this->data = $data;
$this->id = $id;
$this->pet = $pet;
}

public function toMap() {
return array(
'id' => $this->id,
'nome' => $this->nome,
'pet' => $this->pet,
'data' => $this->data
);
}

public static function fromMap($map) {
return new self($map["nome"], $map["data"], $map["id"], $map["pet"]);
}
}

File: lib/models/remedio_model.dart


class Remedio {
  late String nome;
  late String data;
  late int id;
  late int pet;

  Remedio({
    required this.nome,
    required this.data,
    required this.id,
    required this.pet
  });

  Map<String, Object> toMap() {
    return {
      'id': id,
      'nome': nome,
      'pet': pet,
      'data': data
    };
  }

  Remedio.fromMap(Map map) {
    id = map["id"];
    nome = map["nome"];
    pet = map["pet"];
    data = map["data"];
  }
}